GliaWidgets/Sources/CallVisualizer/CallVisualizer.swift

```diff
@@ -81,22 +51,54 @@ public final class CallVisualizer {
                 engagedOperator: environment.engagedOperator
             )
         )
+    }()
 
-        coordinator?.showVisitorCodeViewController(by: presentation)
+    init(environment: Environment) {
+        self.environment = environment
+    }
+
+    deinit {
+        environment.screenShareHandler.status.removeObserver(self)
+        environment.screenShareHandler.cleanUp()
+    }
 
-        startObservingInteractorEvents()
+    /// Show VisitorCode for current Visitor.
+    ///
+    /// Call Visualizer Operators use the Visitor's code to start an Call Visualizer Engagement with the Visitor.
+    ///
+    /// Each Visitor code is generated on demand and is unique for every Visitor on a particular site. Upon the first time
+    /// this function is called for a Visitor the code is generated and returned. For each successive call thereafter the
+    /// same code will be returned as long as the code has not expired. During that time the code can be used to initiate an engagement.
+    /// Once Operator uses the Visitor code to initiate an engagement, the code will expire immediately. When the Visitor Code
+    /// expires this function will return a new Visitor code.
+    ///
+    /// Visitor code can be displayed in 2 ways:
+    /// - `Popup alert`
+    /// - `Embedded view`
+    ///
+    ///  The way of displaying is determined by the 'presentation' parameter. If choosing alert, the current viewController
+    ///  needs to be passed in as an associated value. If, however, embedding is chosen, the view into which you want to
+    ///  embed it has to be passed in as an associated value.
+    public func showVisitorCodeViewController(
+        by presentation: Presentation
+    ) {
+        coordinator.showVisitorCodeViewController(by: presentation)
     }
+}
 
+// MARK: - Internal
```

Comment:
Maybe then the extension can be private? It doesn't seem like any of these methods should be accessible even internally.

Reply:
Methods in that extension are accessed internally, from `Glia` class